connectors/sources/tests/test_oracle.py

```diff
@@ -36,3 +36,14 @@ async def test_ping_negative(patch_logger):
         # Execute
         with pytest.raises(Exception):
             await source.ping()
+
+
+@pytest.mark.asyncio
+@patch("connectors.sources.oracle.create_engine")
+async def test_create_engine(mock_fun):
+    """Test create_engine method of OracleDataSource class"""
+    # Setup
+    source = create_source(OracleDataSource)
+    source.oracle_home = "/home/devuser"
+    mock_fun.return_value = MagicMock()
+    source._create_engine()
```

Comment:
What does this test check? That no exception is thrown?

I can see that the `_create_engine` method has two cases:
- oracle_home is already present -> change the environment variable and create the engine in thick mode
- oracle_home is not present -> just create it from the connection string

I would advocate for having explicit assertions/tests, that the engine is created correctly for both cases. You could check that the passed args for `create_engine` are the expected ones.

Reply:
Added two separate test cases for `_create_engine` and added assert with `assert_called_with`.